pkg/templates/ignition.go

```diff
@@ -95,6 +95,11 @@ func (i *ignition) GenerateNode(kluster *kubernikusv1.Kluster, pool *models.Node
 		}
 	}
 
+	var nodeTaints []string
+	if pool != nil && strings.HasPrefix(pool.Flavor, "zg") {
```

Comment:
I'm not a fan of this. But it's not your problem. We need to expose the node constraints into the flavor properties instead of encoding it into the flavor name. We already have precedent of doing it like this too.